src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            get { return m_IsMatchAppTheme; }
+            set
+            {
+                if (m_IsMatchAppTheme == value)
+                {
+                    return;
+                }
+
+                m_IsMatchAppTheme = value;
+                SetGraphTheme(m_IsMatchAppTheme);
+            }
+        }
+
+        public Style SelectTextBoxStyle(bool incorrectRange, bool error)
+        {
+            if ((incorrectRange || error) && this.Resources.ContainsKey("ErrorTextBoxStyle"))
+            {
+                return (Style)(this.Resources["ErrorTextBoxStyle"]);
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public void SetGrapher(GraphControl.Grapher grapher)
+        {
+            m_ViewModel.SetGrapher(grapher);
+        }
+
+        public void RefreshRanges()
+        {
+            ViewModel.ResetView();
+        }
+
+        public static string GetLineWidthAutomationName(double width)
+        {
+            var resourceLoader = AppResourceProvider.GetInstance();
+
+            if (width == 1.0)
+            {
+                return resourceLoader.GetResourceString("SmallLineWidthAutomationName");
+            }
+            else if (width == 2.0)
+            {
+                return resourceLoader.GetResourceString("MediumLineWidthAutomationName");
+            }
+            else if (width == 3.0)
+            {
+                return resourceLoader.GetResourceString("LargeLineWidthAutomationName");
+            }
+            else
+            {
+                return resourceLoader.GetResourceString("ExtraLargeLineWidthAutomationName");
+            }
+        }
+
+        // Event sends the if the IsMatchAppTheme is selected
+        public event System.EventHandler<bool> GraphThemeSettingChanged;
+
+        private void GridSettingsTextBox_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
+        {
+            if (e.Key == VirtualKey.Enter)
+            {
+                // CSHARP_MIGRATION: TODO:
+                // check if condition
+                if (FocusManager.TryMoveFocusAsync(FocusNavigationDirection.Next).Status != AsyncStatus.Completed)
+                {
+                    IAsyncOperation<FocusMovementResult> result = FocusManager.TryMoveFocusAsync(FocusNavigationDirection.Previous);
+                }
+                e.Handled = true;
+            }
+        }
+
+        private void ResetViewButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            ViewModel.ResetView();
+        }
+
+        private void SetGraphTheme(bool isMatchAppTheme)
+        {
+            string propertyName = isMatchAppTheme ? "IsMatchAppTheme" : "IsAlwaysLightTheme";
+            ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
+            localSettings.Values.Add("IsGraphThemeMatchApp", isMatchAppTheme);
+            GraphThemeSettingChanged(this, isMatchAppTheme);
```

Comment:
check null reference before raising up this event

Reply:
Used ?.Invoke() instead to avoid null reference issue.